src/main/java/de/tum/in/www1/artemis/repository/ComplaintRepository.java

```diff
@@ -54,4 +68,9 @@
      * @return number of complaints associated to exercise exerciseId and tutor tutorId
      */
     long countByResult_Participation_Exercise_IdAndResult_Assessor_Id(Long exerciseId, Long tutorId);
+
+    /**
+     * Delete all complaints that belong to submission results of a given participation
+     */
+    void deleteByResult_Participation_Id(@Param("exerciseId") Long exerciseId);
```

Comment:
Why the param is called `exerciseId`? Shouldn't be `participationId`? And since you don't use a custom query, do you need the `@Param` notation?

Reply:
You are right, in my first approach I used the exerciseId here. I updated it.